include/libtorrent/entry.hpp

```diff
@@ -95,8 +95,28 @@ namespace libtorrent
 			bool operator()(T1 const& rhs, T2 const& lhs) const
 			{ return rhs < lhs; }
 		};
-	};
+
+		template<class T> using map_string = std::map<std::string, T, aux::strview_less>;
+#else
+		template<class T>
+		class map_string : public std::map<std::string, T>
```

Comment:
if you make it a ``struct`` instead you don't have to say ``public`` here nor below (provided you put the public members first)